// react-native-platform-searchbar/lib/commonjs/components/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ButtonIOS = ({
  onPress,
  style: _style,
  accessibilityLabel,
  hitSlop,
  children
}) => {
  return /*#__PURE__*/_react.default.createElement(_reactNative.Pressable, {
    onPress: onPress,
    accessibilityLabel: accessibilityLabel,
    accessibilityComponentType: "button",
    accessibilityRole: "button",
    accessibilityTraits: "button",
    hitSlop: hitSlop,
    style: ({
      pressed
    }) => [_style, pressed && {
      opacity: 0.5
    }]
  }, children);
};

const ButtonAndroid = ({
  onPress,
  style,
  accessibilityLabel,
  hitSlop,
  children
}) => {
  return /*#__PURE__*/_react.default.createElement(_reactNative.Pressable, {
    onPress: onPress,
    style: style,
    accessibilityLabel: accessibilityLabel,
    accessibilityComponentType: "button",
    accessibilityRole: "button",
    accessibilityTraits: "button",
    hitSlop: hitSlop,
    android_ripple: {
      color: '#888',
      borderless: true
    }
  }, children);
};

var _default = _reactNative.Platform.OS === 'ios' ? ButtonIOS : ButtonAndroid;

exports.default = _default;
