// react-native-platform-searchbar/lib/module/components/Button.js
import React from 'react';
import { Pressable, Platform } from 'react-native';

const ButtonIOS = ({
  onPress,
  style: _style,
  accessibilityLabel,
  hitSlop,
  children
}) => {
  return /*#__PURE__*/React.createElement(Pressable, {
    onPress: onPress,
    accessibilityLabel: accessibilityLabel,
    accessibilityComponentType: "button",
    accessibilityRole: "button",
    accessibilityTraits: "button",
    hitSlop: hitSlop,
    style: ({
      pressed
    }) => [_style, pressed && {
      opacity: 0.5
    }]
  }, children);
};

const ButtonAndroid = ({
  onPress,
  style,
  accessibilityLabel,
  hitSlop,
  children
}) => {
  return /*#__PURE__*/React.createElement(Pressable, {
    onPress: onPress,
    style: style,
    accessibilityLabel: accessibilityLabel,
    accessibilityComponentType: "button",
    accessibilityRole: "button",
    accessibilityTraits: "button",
    hitSlop: hitSlop,
    android_ripple: {
      color: '#888',
      borderless: true
    }
  }, children);
};

export default Platform.OS === 'ios' ? ButtonIOS : ButtonAndroid;
